client/extensions/woocommerce/woocommerce-services/state/shipping-label/actions.js

```diff
@@ -487,14 +487,15 @@ export const removePackage = ( orderId, siteId, packageId ) => {
 	};
 };
 
-export const setPackageType = ( orderId, siteId, packageId, boxTypeId ) => {
-	return {
+export const setPackageType = ( orderId, siteId, packageId, boxTypeId ) => ( dispatch, getState ) => {
+	dispatch( {
 		type: WOOCOMMERCE_SERVICES_SHIPPING_LABEL_SET_PACKAGE_TYPE,
 		siteId,
 		orderId,
 		packageId,
 		boxTypeId,
-	};
+		allBoxes: getAllPackageDefinitions( getState(), siteId ),
```

Comment:
Using an action object to pass data down to the reducer feels wrong... at least pass down only `box: allBoxes[boxTypeId]`.